rx-preferences/src/main/java/com/f2prateek/rx/preferences/RxSharedPreferences.java

```diff
@@ -93,10 +93,11 @@ public void onSharedPreferenceChanged(SharedPreferences preferences, String key)
     return new Preference<>(preferences, key, defaultValue, FloatAdapter.INSTANCE, keyChanges);
   }
 
-  /** Create an integer preference for {@code key}. Default is {@code null}. */
+  /** Create an integer preference for {@code key}. Default is {@code 0}. */
   @CheckResult @NonNull
   public Preference<Integer> getInteger(@NonNull String key) {
-    return getInteger(key, null);
+    //noinspection UnnecessaryBoxing
+    return getInteger(key, Integer.valueOf(0));
```

Comment:
I would probably cache them all in `DEFAULT_<foo>` static fields in this class to be explicit. Relying on the caching/interning of instance of Java seems a bit too implicit.